src/components/modules/Authentcation/Register/Register.tsx

```diff
@@ -1,5 +1,191 @@
+              onClick={navigateRegister}
+            >
+              {/* <Link to="/register" className="text-6xl"> */}
+              <i className="fa-solid fa-user-plus text-6xl"></i>
+              {/* </Link> */}
+              <h6 className="font-semibold pt-2">sign up</h6>
+            </div>
+          </div>
+          <form onSubmit={handleSubmit(onsubmit)}>
+            <div className="flex flex-col mdl:flex-row justify-between items-center ">
+              <div>
+                <label htmlFor="first name" className="block">
+                  Your first name
+                </label>
+                <div className="flex items-center border-2 border-gray-600 rounded-3xl px-3 py-2 focus-within:border-auttextcolor2">
+                  <i className="fa-solid fa-address-card bg-authbackground text-xl mr-3"></i>
+
+                  <input
+                    type="text"
+                    placeholder="Type Your first name"
+                    className="  w-full bg-authbackground border-none outline-none"
+                    {...register("first_name", {
+                      required: "firstname is require",
+                    })}
+                  />
+                </div>
+                {errors ? (
+                  <span className="text-red-500">
+                    {errors.first_name?.message}
+                  </span>
+                ) : (
+                  ""
+                )}
+              </div>
+
+              <div>
+                <label htmlFor="last name" className="block">
+                  Your last name
+                </label>
+                <div className="flex items-center border-2 border-gray-600 rounded-3xl px-3 py-2 focus-within:border-auttextcolor2">
+                  <i className="fa-solid fa-address-card bg-authbackground text-xl mr-3"></i>
+
+                  <input
+                    type="text"
+                    placeholder="Type Your last name"
+                    className="  w-full bg-authbackground border-none outline-none"
+                    {...register("last_name", {
+                      required: "lastname is require",
+                    })}
+                  />
+                </div>
+                {errors ? (
+                  <span className="text-red-700">
+                    {errors.last_name?.message}
+                  </span>
+                ) : (
+                  ""
+                )}
+              </div>
+            </div>
+            <div className="py-1">
+              <label htmlFor="email" className="py-2">
+                Your email address
+              </label>
+              <div className="rounded-3xl border-2 flex p-3 border-gray-600 items-center">
+                <i className="fa-solid fa-envelope bg-authbackground text-xl mr-3"></i>
+                <input
+                  type="email"

```

Comment:
We are handling the validations using react hook form. Having the type here will prevent react-hook-form from running the validations in the register function.